src/website/services/domains.service.ts

```diff
@@ -0,0 +1,94 @@
+import { Injectable, Logger, NotFoundException } from '@nestjs/common'
+import { AwsRoute53Service } from 'src/aws/services/awsRoute53.service'
+import { WebsiteService } from './website.service'
+import { WebsiteStatus } from '@prisma/client'
+import { OperationStatus } from '@aws-sdk/client-route-53-domains'
+
+@Injectable()
+export class DomainsService {
+  private readonly logger = new Logger(DomainsService.name)
+
+  constructor(
+    private readonly route53: AwsRoute53Service,
+    private readonly website: WebsiteService,
+  ) {}
+
+  async getDomainDetails(domainName: string) {
+    return this.route53.getDomainDetails(domainName)
+  }
+
+  async searchForDomain(domainName: string) {
+    const tld = domainName.split('.').at(-1)
+
+    const [pricesResp, availabilityResp] = await Promise.all([
+      this.route53.listPrices(tld),
+      this.route53.checkDomainAvailability(domainName),
+    ])
```

Comment:
May be good to wrap in a try-catch in case one of these throws and then we can throw a `BadGatewayException`?